### PR TITLE
translate-c: Don't make const parameters mutable.

### PR DESCRIPTION
**Pull request by fweig**
My attempt at fixing #4106. Whats still missing here is removing the outer block when all parameters are const.

### CODE REVIEW

File: src/zig_clang.cpp [hunk 1]
```diff
@@ -1625,6 +1625,10 @@ unsigned ZigClangFunctionDecl_getAlignedAttribute(const struct ZigClangFunctionD
     return 0;
 }
 
+ZigClangQualType ZigClangParmVarDecl_getType(const struct ZigClangParmVarDecl *self) {
```
Comment (andrewrk) [inline src/zig_clang.cpp]: should be named `ZigClangParmVarDecl_getOriginalType` after the function that it is calling

**Comment by fweig** [comment]
@andrewrk sure thing, that looks very straightforward. How do i run the tests tho? I tried `./build/zig build test` but that reports only passes even if i add a test that must fail.

**Comment by LemonBoy** [comment]
You probably want to run `test-run-translated-c` and `test-translate-c`.

**Comment by andrewrk** [comment]
There are some instructions here on how to run the tests: https://github.com/ziglang/zig/blob/master/CONTRIBUTING.md#improving-translate-c

**Comment by andrewrk** [comment]
> @andrewrk sure thing, that looks very straightforward. How do i run the tests tho? I tried `./build/zig build test` but that reports only passes even if i add a test that must fail.

That should not be the case. Did you wait the full hour(s) for the test suite before drawing that conclusion? (I don't recommend to do that; if you do what @LemonBoy said above that should be sufficient testing)

**Comment by fweig** [comment]
ok nvm, it's working as intended. Thanks for both of your help! Just added a unittest and renamed the wrapper function.

**Comment by andrewrk** [comment]
Well done!

**Pull request merged**